### ISSUE TITLE
Native <track> Subtitles / Closed-captions not rendering in IE 11, Edge, Firefox

### ISSUE THREAD

**Issue by shanepoints** [videojs/video.js#5191]
## Description
I am using Player.addRemoteTextTrack() to include a vtt file for closed captions.  In Chrome/Safari/etc this adds the captions as expected, however it seems to silently fail in IE 11 and MS Edge.  I have created a very simple example here: https://jsfiddle.net/terpfhgq/7/

## Steps to reproduce
Explain in detail the exact steps necessary to reproduce the issue.
1.  Create an instance of VideoJS player
2.  Add a vtt caption file via Player.addRemoteTextTrack()
3.  Open in IE 11/Edge (and open in Chrome/Safari for comparison)

### Expected
The expected result is what happens in Chrome: the caption button displays in the controlbar and clicking the relevant option causes the closed captions to render.

### Actual
In IE 11 and Edge the caption button does not display, and there is no way for the user to retrieve the vtt file and render the closed captions text.

### Error output
N/A (silent failure)

### versions
Windows 10 pro, Version 1709
Internet Explorer 11.431.16299.0
MS Edge 41.16299.402.0
#### videojs
Version 6.8.0 (This is the latest official release, but I have tested several of the more recent pre-releases up to 7.0.0 with the same results)
#### browsers
IE 11, MS Edge
#### OSes
Windows 10 pro
### plugins
N/A

**Comment by shanepoints** [comment]
It seems like this is a [known issue with the dash plugin](https://github.com/videojs/videojs-contrib-dash/issues/191), however, in my use case the plugin is not present.

**Comment by gkatsev** [comment]
Well, I was starting to write out one answer but seems like it's no longer the necessarily the case.
I believe IE11 is super strict and will not load in text tracks that aren't served with the `text/vtt` mimetype. Your example serves it with `text/plain`. Chrome loads it fine but it does show a warning in the console. Seems like MS Edge will also load it. I got it to work by adding the `default: true` option to `addRemoteTextTrack`. I'm guessing the reason why it isn't showing up in the menu is because Edge and IE11 don't properly trigger `addtrack` events when `<track>` elements are added.
Is there a reason why you're using native text tracks?

**Comment by gkatsev** [comment]
http://output.jsbin.com/zihunacowu/2 is your example in jsbin, because jsfiddle was having issues initially. https://jsfiddle.net/terpfhgq/8/ has `default: true`

**Comment by gkatsev** [comment]
Yeah, firefox also has had issues related to it. That's Chrome works decently well but that's why by default we use emulated captions everywhere except iOS and Safari.

The WCAG thing is interesting, though, I think, and hope, they care mostly about having captions and subtitles capabilities rather than specifically using the `<track>` element.

@OwenEdwards do you have any thoughts about that?

**Comment by OwenEdwards** [comment]
The relevant WCAG Success Criterion is **1.2.2** which requires that:

>  Captions are provided for all prerecorded audio content in synchronized media ...

The Understanding WCAG 2.0 section [Understanding SC 1.2.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/media-equiv-captions.html) is helpful in expanding on that Success Criterion.

What @shanepoints links to ([H95: Using the track element to provide captions](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/H95)) is a "**Technique**" (specifically a "**Sufficient Technique**"), but as the Understanding WCAG 2.0 section points out, H95:

> ... represents a technique or combination of techniques that the WCAG Working Group deems sufficient for meeting this Success Criterion. However, it is not **necessary** to use these particular techniques.

In other words, using a native captions track in HTML meets the requirement, but is not the *only* way to meet the requirement. Other techniques that cause captions to be available (even burned-in open-captions) can be sufficient.

**Comment by shanepoints** [comment]
@gkatsev It seems reasonable to close since there is a clear path to compliance.  Does this mean that in the future VideoJS will phase out native text tracks?

**Comment by gkatsev** [comment]
I don't think we'll ever remove support for native text tracks but depending on how browsers change their native webvtt support, we may end up de-emphasizing native support even more than we already do. Hopefully, as browsers improve their native capabilities, these issues will go away.

Though, some of the issues are related to how Video.js currently handles the video element in the embed code and it's possible things will work slightly better if you use the `<video-js>` embed code instead.

I'm going to close this issue but we can continue discussion here if we find new information or things come up.

**Comment by arski** [comment]
@practual may be relevant to your change
